fabrics.c

```diff
@@ -665,7 +735,8 @@ int nvmf_discover(const char *desc, int argc, char **argv, bool connect)
 		nvme_free_tree(r);
 		return ret;
 	}
-	nvme_read_config(r, config_file);
+	if (!nvme_read_config(r, config_file))
+		json_config = true;
```

Comment:
I am still pondering if the `json_config` makes sense. `nvme_read_config` returns 0 independent if the config file exist or not. But we just fail if `nvme_read_config()` returns an error I suppose.

Reply:
I had a discussion offline with @hreinecke and @martin-gpy and we decided this approach is okay.